qiskit/algorithms/state_fidelities/base_state_fidelity.py

```diff
@@ -0,0 +1,264 @@
+           circuits_1: (Parametrized) quantum circuits preparing the
+                        first list of quantum states.
+           circuits_2: (Parametrized) quantum circuits preparing the
+                        second list of quantum states.
+           values_1: Numerical parameters to be bound to the first circuits.
+           values_2: Numerical parameters to be bound to the second circuits.
+
+        Returns:
+             List of parameter values for fidelity circuit.
+
+        """
+
+        values_1 = self._preprocess_values(circuits_1, values_1)
+        values_2 = self._preprocess_values(circuits_2, values_2)
+
+        values = []
+        if values_2 is not None or values_1 is not None:
+            if values_2 is None:
+                values = values_1
+            elif values_1 is None:
+                values = values_2
+            else:
+                for (val_1, val_2) in zip(values_1, values_2):
+                    values.append(val_1 + val_2)
+
+        return values
+
+    @abstractmethod
+    def _run(
+        self,
+        circuits_1: Sequence[QuantumCircuit],
+        circuits_2: Sequence[QuantumCircuit],
+        values_1: Sequence[Sequence[float]] | None = None,
+        values_2: Sequence[Sequence[float]] | None = None,
+        **run_options,
+    ) -> StateFidelityResult:
+        """Compute the state overlap (fidelity) calculation between 2
+        (parametrized) circuits (first and second) for a specific set of parameter
+        values (first and second). This calculation depends on the particular
+        fidelity method implementation.
+        Args:
+            circuits_1: (Parametrized) quantum circuits preparing one set of states
+            circuits_2: (Parametrized) quantum circuits preparing another set of states
+            values_1: Numerical parameters to be bound to the first circuits
+            values_2: Numerical parameters to be bound to the second circuits.
+            run_options: Backend runtime options used for circuit execution.
+
+        Returns:
+            The result of the fidelity calculation.
+        """
+        ...
+
+    def run(
+        self,
+        circuits_1: Sequence[QuantumCircuit],
+        circuits_2: Sequence[QuantumCircuit],
+        values_1: Sequence[Sequence[float]] | None = None,
+        values_2: Sequence[Sequence[float]] | None = None,
+        **run_options,
+    ) -> PrimitiveJob:
+        r"""
+        Run asynchronously the state overlap (fidelity) calculation between 2
+        (parametrized) circuits (left and right) for a specific set of parameter
+        values (left and right).
+
+        Args:
+            circuits_1: (Parametrized) quantum circuits preparing :math:`|\psi\rangle`.
+            circuits_2: (Parametrized) quantum circuits preparing :math:`|\phi\rangle`.
+            values_1: Numerical parameters to be bound to the left circuits.
+            values_2: Numerical parameters to be bound to the right circuits.
```

Comment:
Since this discussion uses left and right and in the values, would it be clearer to have `left_circuit`, `right_circuit`, `left_values` and `right_values` or just drop the notion of left and right here?

Reply:
I think that I should drop the notion of left and right because as Stefan pointed out, not all methods that could inherit from this base class use it.